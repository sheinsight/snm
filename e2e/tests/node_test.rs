use std::env::current_dir;

use e2e::SnmEnv;

e2e::assert_snapshot! {
  #[tokio::test]
  test_auto_install_node_with_node_version,
  cwd: current_dir()?.join("tests").join("fixtures").join("auto_install_node_with_node_version"),
  envs: vec![],
  commands: [
    "node -v",
  ]
}

e2e::assert_snapshot! {
  #[tokio::test]
  test_show_node_version_with_strict_mode,
  cwd: current_dir()?.join("tests").join("fixtures").join("empty"),
  envs: vec![SnmEnv::Strict("true".to_string())],
  commands: [
    "node -v",
  ]
}

e2e::assert_snapshot! {
  #[tokio::test]
  test_no_strict_and_no_default_node,
  cwd: current_dir()?.join("tests").join("fixtures").join("empty"),
  envs: vec![SnmEnv::Strict("false".to_string())],
  commands: [
    "node -v",
  ]
}

e2e::assert_snapshot! {
  #[tokio::test]
  test_with_strict_mode_and_has_default_node,
  cwd: current_dir()?.join("tests").join("fixtures").join("empty"),
  envs: vec![SnmEnv::Strict("true".to_string())],
  commands: [
    "snm node install 20.0.0",
    "snm node list --compact",
    "snm node default 20.0.0",
    "snm node list --compact",
    "node -v",
  ]
}
