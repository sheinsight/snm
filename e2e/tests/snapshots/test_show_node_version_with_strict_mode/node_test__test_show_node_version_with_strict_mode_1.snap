---
source: e2e/tests/node_test.rs
expression: res
snapshot_kind: text
---
is: node -v
os: Error: In strict mode, a .node-version file must be configured in the current directory.
